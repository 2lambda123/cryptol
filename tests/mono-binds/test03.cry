module test03 where

test a = f a
  where
  f x = g x
  g x = f x
