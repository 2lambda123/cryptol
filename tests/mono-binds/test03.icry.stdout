Loading module Cryptol
Loading module Cryptol
Loading module test03
module test03
import Cryptol
/* Not recursive */
test03::test : {a, b} a -> b
test03::test = \{a, b} (a : a) ->
               f a b a
               where
                 /* Recursive */
                 f : {c, d} c -> d
                 f = \{c, d} (x : c) -> g c d x
                 g : {c, d} c -> d
                 g = \{c, d} (x : c) -> f c d x
                 
               

Loading module Cryptol
Loading module test03
module test03
import Cryptol
/* Not recursive */
test03::test : {a, b} a -> b
test03::test = \{a, b} (a : a) ->
               f a b a
               where
                 /* Recursive */
                 f : {c, d} c -> d
                 f = \{c, d} (x : c) -> g c d x
                 g : {c, d} c -> d
                 g = \{c, d} (x : c) -> f c d x
                 
               

